login_service/sparql_queries.rb

```diff
@@ -31,8 +31,50 @@ def insert_new_session_for_account(account, session_uri, session_id, membership_
       Mu::AuthSudo.update(query)
     end
 
+    def insert_login_activity(user)
+      now = DateTime.now
+
+      query = %(
+      SELECT ?login_activity
+      WHERE {
+        GRAPH <#{SYSTEM_USERS_GRAPH}> {
+          ?login_activity a <#{MU_EXT.LoginActivity}> ;
+            <#{PROV.wasAssociatedWith}> <#{user}> .
+        }
+      }
+      LIMIT 1)
+
+      result = Mu::AuthSudo.query(query)
+      unless result.empty?
+        login_activity = result.first[:login_activity].to_s
+
+        # Delete old login activity
+        query = %(
+        DELETE WHERE {
+          GRAPH <#{SYSTEM_USERS_GRAPH}> {
+            <#{login_activity}> ?p ?o .
+          }
+        })
+        Mu::AuthSudo.update(query)
+      end
```

Comment:
Just one `DELETE WHERE` query deleting all `ext:LoginActivity` associated with the user? No need to do the `SELECT` first